hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -147,65 +159,50 @@ func virtualMachineTemplateBase(image string, kvPlatform *hyperv1.KubevirtNodePo
 		},
 	}
 
+	if kvPlatform.RootVolume != nil {
+		if kvPlatform.RootVolume.Persistent != nil && kvPlatform.RootVolume.Persistent.Size != nil {
+			volumeSize = *kvPlatform.RootVolume.Persistent.Size
+		}
+
+		if kvPlatform.RootVolume.Persistent != nil {
+			storageClass = kvPlatform.RootVolume.Persistent.StorageClass
+
+			for _, am := range kvPlatform.RootVolume.Persistent.AccessModes {
+				accessModes = append(accessModes, corev1.PersistentVolumeAccessMode(am))
+			}
+
+			if len(accessModes) == 0 {
+				accessModes = append(accessModes, corev1.ReadWriteOnce)
+			}
+		}
+	}
+
 	dataVolume := kubevirtv1.DataVolumeTemplateSpec{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
 			Name: rootVolumeName,
 		},
-	}
-
-	if strings.HasPrefix(image, "docker://") {
-		dataVolume.Spec = v1beta1.DataVolumeSpec{
-			Source: &v1beta1.DataVolumeSource{
-				Registry: &v1beta1.DataVolumeSourceRegistry{
-					URL:        &image,
-					PullMethod: &pullMethod,
-				},
-			},
-		}
-	} else {
-		dataVolume.Spec = v1beta1.DataVolumeSpec{
-			Source: &v1beta1.DataVolumeSource{
-				HTTP: &v1beta1.DataVolumeSourceHTTP{
-					URL: image,
+		Spec: v1beta1.DataVolumeSpec{
+			Source: dvSource,
+			Storage: &v1beta1.StorageSpec{
+				AccessModes: accessModes,
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]apiresource.Quantity{
+						corev1.ResourceStorage: volumeSize,
+					},
 				},
-			},
-		}
-	}
-
-	dataVolume.Spec.Storage = &v1beta1.StorageSpec{
-		Resources: corev1.ResourceRequirements{
-			Requests: map[corev1.ResourceName]apiresource.Quantity{
-				corev1.ResourceStorage: volumeSize,
+				StorageClassName: storageClass,
 			},
 		},
 	}
 
-	if kvPlatform.RootVolume != nil &&
-		kvPlatform.RootVolume.Persistent != nil {
-		if kvPlatform.RootVolume.Persistent.StorageClass != nil {
-			dataVolume.Spec.Storage.StorageClassName = kvPlatform.RootVolume.Persistent.StorageClass
-		}
-		if len(kvPlatform.RootVolume.Persistent.AccessModes) != 0 {
-			var accessModes []corev1.PersistentVolumeAccessMode
-			for _, am := range kvPlatform.RootVolume.Persistent.AccessModes {
-				amv1 := corev1.PersistentVolumeAccessMode(am)
-				accessModes = append(accessModes, amv1)
-			}
-			dataVolume.Spec.Storage.AccessModes = accessModes
-		}
-	}
```

Comment:
@nunnatsa this is what i mean about the storageClass and accessModes only being defined on the DV when the user provides them. The previous logic worked this way on purpose.